src/cache/http_cache_provider.cpp

```diff
@@ -198,10 +199,11 @@ void http_cache_provider_t::set_data(const std::string& key, const std::string&
   http::Request request(url);
   http::Response response = request.send(method, data, http_header);
 
-  // A successful response must have the code 200 or 201.
-  if (response.status != http::Response::Ok && response.status != http::Response::Created) {
+  // A successful response must have the code 200 or 201. Or 204, in that case do not try to read the response body.
+  if (response.status != http::Response::Ok && response.status != http::Response::Created && response.status != http::Response::NoContent) {
     std::ostringstream ss;
-    ss << "HTTP remote responded (" << response.status << "): " << response.body.data()
+    ss << "HTTP remote responded (" << response.status << "): " 
+       << (response.body.empty() ? reinterpret_cast<const uint8_t*>("") : response.body.data())
```

Comment:
Is this `empty()` check necessary, now that you filter out `http::Response::NoContent`?

Reply:
Yes, because otherwise it crashes if the server does not send a body. `std::vector::data()` must not be dereferenced if empty.

Reply:
Thanks @gjasny for explanation.